### DISCUSSION TITLE
Kmac from python

### DISCUSSION THREAD

**Discussion by Mysfrit** [XKCP/XKCP#95] [Q&A] [answered]
Hi, 
currently, I'm working on a small project written in python. I would love to use KMAC from this library. I've seen a few uses of calling C library from python and I want to ask if anyone got any experience with this or if it's even doable. There ain't any implementation for python so this is my only choice.
Thanks

**Comment by gvanas** [comment, marked as answer]
Personally I don't have any experience with calling C from Python.

Even though KMAC is not supported by Python's standard library, a minor interface adjustment would open Python's internal implementation to cSHAKE (on which KMAC is based). I introduced an [issue](https://bugs.python.org/issue39539), but haven't been actively pushing for it. Perhaps if you add a comment, they will know that there is still some interest for it…
